app/controlplane/pkg/metrics/prometheus/collector.go

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheuscollector
+
+import (
+	"slices"
+
+	"github.com/go-kratos/kratos/v2/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ChainloopCollector is an interface for a collector that collects metrics for Chainloop
+// It extends the prometheus.Collector interface plus two additional methods
+type ChainloopCollector interface {
```

Comment:
Is this interface been used?

Reply:
Hmm apparently not, removing it, thanks!